src/cli/common/scripts-info.ts

```diff
@@ -99,3 +100,14 @@ const _scripts = {
 
 export const scripts = _scripts as Record<keyof typeof _scripts, ScriptInformation>
 
+export function getValidOptionsForCompletion(options: OptionDefinition[], prevArgs: string[]): string[] {
+	return options
+		.filter(o => o.multiple || !prevArgs.includes(asOptionName(o.name)) && (!o.alias || !prevArgs.includes(asOptionName(o.alias))))
```

Comment:
i think we can make this more readable by outsourcing the filter to a function with a nice name!